### DISCUSSION TITLE
Prise en charge DB Local

### DISCUSSION THREAD

**Discussion by MartiNu-dev** [Issaminu/skyline-app#3] [General]
Bonjour,

Quel coût estimez-vous la prise en charge d'une base de données locale, type on-prem, comme du SQLite ou MariaDB par exemple ?
Ce serait plus simple à déployer lors d'une install locale que de devoir déployer un serveur S3, Auth0 etc..

**Comment by Issaminu** [comment]
**Le déploiement de votre base de données en local serait pratiquement gratuit** puisque vous n'utilisez pas de service en nuage.
Les coûts d'une base de données proviennent de l'utilisation d'un DBaaS (DataBase as a Service) tel que Amazon RDS ou PlanetScale, ces services sont facturés en fonction de l'utilisation, plus le nombre d'opérations de lecture/écriture par mois est élevé, plus la facture est élevée. [Voici un exemple, il s'agit de la te tarification de PlanetScale](https://planetscale.com/pricing).

Si vous prévoyez de déployer votre base de données localement, alors elle sera gratuite, puisque MySQL/MariaDB/SQlite sont tous des systèmes de gestion de base de données gratuits.

Je ne suis pas sûr de comprendre ce que vous signifiez par l'utilisation d'une base de données locale au lieu d'utiliser S3 et Auth0, car aucun d'entre eux n'est un DBaaS, et ils répondent tous deux à des objectifs différents.

Amazon S3 est un service de stockage d'objets, utilisé pour stocker des fichiers et les servir de manière programmatique. J'utilise S3 dans cette application pour servir les images des utilisateurs à partir du nuage, et non comme un service de base de données. Si vous ne souhaitez pas utiliser S3, vous pouvez simplement stocker les images localement dans l'application elle-même, dans le répertoire `public`. Cela vous permettra de servir les images directement depuis le serveur d'application lui-même, sans avoir besoin de connecter avec un service externe.
Le problème qui se pose est que vous ne pouvez pas servir les images qui sont enregistrées sur le serveur a une moment d'exécution (run-time), mais seulement avans la construction (build-time). [Il ne s'agit pas d'une limitation de l'application elle-même, mais de NextJS (veuillez lire la note en bas de page)](https://nextjs.org/docs/basic-features/static-file-serving#:~:text=note%3A%20only%20assets%20that%20are%20in%20the%20public%20directory%20at%20build%20time%20will%20be%20served%20by%20next.js.%20files%20added%20at%20runtime%20won't%20be%20available.%20we%20recommend%20using%20a%20third%20party%20service%20like%20aws%20s3%20for%20persistent%20file%20storage.). Si cela ne correspond pas à votre cas d'utilisation, il est alors nécessaire d'utiliser un service de stockage d'objets tel que Amazon S3 ou Cloudinary.

En ce qui concerne Auth0, c'est une plateforme d'autorisation, je l'ai utilisé pour gérer l'authentification des utilisateurs dans l'application. Si vous préférez gérer l'authentification localement, vous pouvez créer votre propre solution ou utiliser une bibliothèque comme NextAuth.

J'espère avoir répondu à votre question.

**Comment by MartiNu-dev** [comment]
Merci pour votre réponse aussi rapide que précise !
J'ai pu me déployer un S3 local et me faire un compte Auth0 afin de récupérer les secrets nécessaires. Par contre, QUID de la base de données applicative ? Vous utilisez prisma, en référence à votre réponse : pourrais-je simplement remplacer l'url de la base prisma par celle d'un MariaDB par exemple ? Auriez vous un exemple de l'URL de connexion ?

Merci pour votre réponse,
Très cordialement,
Martin
